tests/integration/replication.tcl

```diff
@@ -424,17 +426,13 @@ test {slave fails full sync and diskless load swapdb recovers it} {
             # for faster server shutdown, make rdb saving fast again (the fork is already uses the slow one)
             $master config set rdb-key-save-delay 0
 
-            # waiting slave to do flushdb (key count drop)
-            wait_for_condition 50 100 {
-                2000 != [scan [regexp -inline {keys\=([\d]*)} [$slave info keyspace]] keys=%d]
-            } else {
-                fail "Replica didn't flush"
-            }
+            # waiting replica to start loading
+            wait_for_log_messages -1 {"*Loading DB in memory*"} $loglines 50 100
 
             # make sure we're still loading
             assert_equal [s -1 loading] 1
 
-            # kill the slave connection on the master
+            # kill the replica connection on the master
             set killed [$master client kill type slave]
 
             # wait for loading to stop (fail)
```

Comment:
this wait is now ineffective.
we need another way to check.
maybe look at `master_sync_in_progress` or set `replicaof no one` and look at `role`?

Reply:
What happens?
Because this test still runs under and test swapdb on sync loading mode

Reply:
ohh, right, no async loading because replid changed.
i saw you removed this code, and concluded it's an async loading:
```tcl
            # waiting slave to do flushdb (key count drop)
            wait_for_condition 50 100 {
                2000 != [scan [regexp -inline {keys\=([\d]*)} [$slave info keyspace]] keys=%d]
```

so a recap, we now have:
1. either disk-based or disk-less that's not `swapdb`, in which case the loading flag is set, and INFO KEYSPACE will be emptied.
2. diskless swapdb when replid changed or modules with issues are present, in which case loading flag is set, and keyspace is NOT emptied.
3. diskless swapdb when replid didn't change and no modules with issues, in which case loading flat is off, and keyspace is NOT emptied.

i.e. before this change we had case 1 (which was also the case when swapdb was used), and no such distinction between the loading flag and the keyspace.

bottom line, the change you made in the test was only needed for the keyspace, not the loading flag. so the test is fine.